File: frontend/src/services/analytics.ts
import { apiClient } from "./api";
import type { AnalyticsSummary } from "../types/analytics";

export const analyticsService = {
  async summary(ownerId: number) {
    const { data } = await apiClient.get<AnalyticsSummary>("/analytics/summary", {
      params: { owner_id: ownerId }
    });
    return data;
  }
};

File: backend/app/api/v1/endpoints/analytics.py
from fastapi import APIRouter, Depends, Query

from app.api.dependencies import get_analytics_service
from app.application.services.analytics import AnalyticsService
from app.schemas.analytics import AnalyticsSummary

router = APIRouter()


@router.get("/summary", response_model=AnalyticsSummary)
async def get_summary(
    owner_id: int = Query(..., description="User identifier"),
    service: AnalyticsService = Depends(get_analytics_service),
) -> AnalyticsSummary:
    return await service.summary_for_owner(owner_id)
